src/components/CreateBlog.vue

```diff
@@ -0,0 +1,195 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import { useRouter } from 'vue-router';
+import { postBlog } from '@/composables/postBlog';
+
+const props = defineProps<{
+  author: {
+    id: string;
+    name: string;
+    image: string;
+  }
+}>();
+
+const category = ref('');
+const title = ref('');
+const description = ref('');
+const image = ref('');
+const router = useRouter();
+const { error, post } = postBlog();
+
+const handleSubmit = async () => {
+  const blogData = {
+    category: category.value,
+    title: title.value,
+    description: description.value,
+    created_at: Math.floor(Date.now() / 1000),
+    author: props.author,
+    image: image.value,
+  };
+
+  await post('Blog', blogData);
+
+  if (error.value) {
+    console.error('Failed to create blog:', error.value);
+  } else {
+    router.push({ name: 'home' });
+  }
```

Comment:
This reactively broken code is very hard to follow. A function can always return a tuple. Hence we should follow better API design.

```javascript
// some-composable.vue

export async function createBlog(payload){
  let res = null
  let err = null

  try{
    res = await this.$api.post("/some-uri", payload)
  }catch(e){
    err = e
  }finally{
    return [res, err]
  }
}

// CreateBlog.vue

const [res, err] = await createBlogPost(payload)

if(err){
  // Notify user about the error.
  
  return
}

router.push({
  name: 'home'
})
```